P0: mypy nibabel.load attr-defined error in profiler.py

## Bug

`mypy` reports `attr-defined` error on `nib.load()` in `src/minivess/data/profiler.py:95`:

```
src/minivess/data/profiler.py:95:11: error: Module "nibabel" does not explicitly export attribute "load"  [attr-defined]
```

And `img.dataobj` at line 96:
```
src/minivess/data/profiler.py:96:35: error: "FileBasedImage" has no attribute "dataobj"  [attr-defined]
```

## Root Cause

nibabel's type stubs are incomplete. `nib.load()` returns `FileBasedImage` which doesn't expose `dataobj` in its type signature.

## Fix

Either:
1. Add `# type: ignore[attr-defined]` comments
2. Use a cast: `img = cast(nib.nifti1.Nifti1Image, nib.load(...))`
3. Wait for upstream nibabel type stub improvements

## Priority

P0 — pre-existing, blocks clean `mypy` runs across the project.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -66,6 +66,8 @@ dependencies = [
     "skan>=0.13.1",
     "psutil>=7.2.2",
     "nvidia-ml-py>=13.590.48",
+    # grpcio 1.78.1 was yanked (gcloud outage). Exclude it explicitly.
+    "grpcio>=1.78.0,!=1.78.1",
 ]
 
 [project.optional-dependencies]

--- a/src/minivess/observability/system_info.py
+++ b/src/minivess/observability/system_info.py
@@ -12,12 +12,8 @@ from __future__ import annotations
 import logging
 import platform
 import subprocess
-from pathlib import Path
 
 logger = logging.getLogger(__name__)
-
-# Default location of the DVC tracking file for the MiniVess dataset
-_DVC_FILE_PATH = Path(__file__).resolve().parents[3] / "data" / "minivess.dvc"
 
 
 def get_system_params() -> dict[str, str]:
@@ -228,11 +224,11 @@ def get_git_info() -> dict[str, str]:
 
 
 def get_dvc_info() -> dict[str, str]:
-    """Collect DVC version and dataset data hash.
+    """Collect DVC version only.
 
-    Reads the DVC tracking file (``data/minivess.dvc``) to extract the
-    content-addressed hash and file count.  Returns ``"unknown"`` for
-    data fields when the file is absent.
+    DVC data hash and nfiles are *identifiers*, not hyperparameters, and
+    belong in MLflow tags (logged via ``log_dvc_provenance()``), not in
+    ``mlflow.log_params()``.  See issue #108.
     """
     info: dict[str, str] = {}
 
@@ -243,22 +239,6 @@ def get_dvc_info() -> dict[str, str]:
         info["sys_dvc_version"] = dvc.__version__
     except (ImportError, AttributeError):
         info["sys_dvc_version"] = "not_installed"
-
-    # Parse .dvc file for md5 hash and nfiles
-    info["sys_dvc_data_hash"] = "unknown"
-    info["sys_dvc_data_nfiles"] = "unknown"
-
-    try:
-        if _DVC_FILE_PATH.exists():
-            text = _DVC_FILE_PATH.read_text(encoding="utf-8")
-            for line in text.splitlines():
-                stripped = line.strip()
-                if stripped.startswith("- md5:"):
-                    info["sys_dvc_data_hash"] = stripped.split(":", 1)[1].strip()
-                elif stripped.startswith("nfiles:"):
-                    info["sys_dvc_data_nfiles"] = stripped.split(":", 1)[1].strip()
-    except (OSError, PermissionError):
-        logger.warning("Could not read DVC file: %s", _DVC_FILE_PATH)
 
     return info
 

--- a/tests/v2/unit/test_system_info.py
+++ b/tests/v2/unit/test_system_info.py
@@ -5,13 +5,9 @@ RED phase: all tests written before implementation exists.
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
 from unittest.mock import MagicMock, patch
 
 import pytest
-
-if TYPE_CHECKING:
-    from pathlib import Path
 
 
 def _gpu_available() -> bool:
@@ -244,13 +240,25 @@ class TestGetGitInfo:
 class TestGetDvcInfo:
     """Tests for get_dvc_info()."""
 
-    def test_returns_dict_with_expected_keys(self) -> None:
+    def test_returns_only_version_key(self) -> None:
+        """get_dvc_info() should return ONLY sys_dvc_version.
+
+        DVC data hash and nfiles are identifiers, not hyperparameters.
+        They belong in MLflow tags (via log_dvc_provenance), not params.
+        Regression test for #108.
+        """
         from minivess.observability.system_info import get_dvc_info
 
         result = get_dvc_info()
         assert "sys_dvc_version" in result
-        assert "sys_dvc_data_hash" in result
-        assert "sys_dvc_data_nfiles" in result
+        assert "sys_dvc_data_hash" not in result, (
+            "sys_dvc_data_hash should not be in get_dvc_info() — "
+            "it's an identifier, not a hyperparameter (#108)"
+        )
+        assert "sys_dvc_data_nfiles" not in result, (
+            "sys_dvc_data_nfiles should not be in get_dvc_info() — "
+            "it's an identifier, not a hyperparameter (#108)"
+        )
 
     def test_all_values_are_strings(self) -> None:
         from minivess.observability.system_info import get_dvc_info
@@ -266,16 +274,17 @@ class TestGetDvcInfo:
         result = get_dvc_info()
         assert result["sys_dvc_version"] != "not_installed"
 
-    def test_graceful_when_no_dvc_file(self, tmp_path: Path) -> None:
-        """When .dvc file doesn't exist, returns 'unknown' for data hash."""
-        from minivess.observability.system_info import get_dvc_info
+    def test_get_all_system_info_excludes_dvc_hash(self) -> None:
+        """get_all_system_info() must not contain DVC hash/nfiles.
 
-        with patch(
-            "minivess.observability.system_info._DVC_FILE_PATH",
-            tmp_path / "nonexistent.dvc",
-        ):
-            result = get_dvc_info()
-        assert result["sys_dvc_data_hash"] == "unknown"
+        Regression test for #108: these are identifiers logged as MLflow
+        tags via log_dvc_provenance(), not as params.
+        """
+        from minivess.observability.system_info import get_all_system_info
+
+        result = get_all_system_info()
+        assert "sys_dvc_data_hash" not in result
+        assert "sys_dvc_data_nfiles" not in result
 
 
 class TestGetAllSystemInfo:

